source/extensions/common/wasm/context.cc

```diff
@@ -1691,8 +1691,14 @@ WasmResult Context::sendLocalResponse(uint32_t response_code, std::string_view b
       if (local_reply_sent_) {
         return;
       }
+      // In case -1 is send from sdk, as an invalid value, nullopt is sent to allows envoy to
+      // set the grpc status code
```

Comment:
I'd rephrase it slightly to mention that C++ and Rust SDKs (and maybe others?) use `-1` (`InvalidCode`) as the default value in case gRPC status is not set, which should be mapped to `nullopt` in Envoy to prevent it from sending `grpc-status` trailer at all.